graphql/resolvers.go

```diff
@@ -3952,7 +3952,7 @@ func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion)
 				if err != nil {
 					return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch version for patch: %s ", err.Error()))
 				}
-				if cpVersion.Aborted {
+				if cpVersion != nil && cpVersion.Aborted {
```

Comment:
This should be a separate check. What does it mean if the version is nil?

Reply:
Does it mean the version we are passing into #L3951 is nil?

Reply:
> This should be a separate check. What does it mean if the version is nil?

The version is nil if the child patch doesn't have a corresponding version. I don't mind making a separate check but I'm under the impression that the condition will short circuit if cpVersion is nil.

Reply:
i was wondering if we are passing a nil value into the db query above and if we should avoid the query all together if we are.

Reply:
> Does it mean the version we are passing into #L3951 is nil?

The case we found in prod showed that cp wasn't nil and that the nil pointer dereference occured on line 3955. https://mongodb.splunkcloud.com/en-US/app/search/search?earliest=%40d&latest=now&q=search%20index%3Devergreen%20metadata.level%20%3E%3D%2070%7C%20spath%20stack%20%7C%20search%20stack%3D%22goroutine%2026266168%20%5Brunning%5D%3A%0Aruntime%2Fdebug.Stack(0x33d0140%2C%200xc00c520bd0%2C%200x3a50cc8)%0A%09runtime%2Fdebug%2Fstack.go%3A24%20%2B0x9f%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.Handler.func1(0x41b71a0%2C%200xc00ba9de60%2C%200x3376060%2C%200x5aabf70%2C%200x0%2C%200xc00793f760)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fhttp_handler.go%3A35%20%2B0x1ad%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql.(*OperationContext).Recover(0xc00ab2ad00%2C%200x41b71a0%2C%200xc00ba9de60%2C%200x3376060%2C%200x5aabf70%2C%200x41b71a0%2C%200xc00ba9de60)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Fcontext_operation.go%3A110%20%2B0x56%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.(*executionContext)._Version_status.func1(0xc00dd537c0%2C%200xc002ac3630%2C%200xc00793ff50)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fgenerated.go%3A37046%20%2B0x7d%0Apanic(0x3376060%2C%200x5aabf70)%0A%09runtime%2Fpanic.go%3A965%20%2B0x1b9%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.(*versionResolver).Status(0xc0115460d0%2C%200x41b71a0%2C%200xc00ba9de60%2C%200xc00f22ba40%2C%200x40a2000%2C%200x2ff1340%2C%200xc007b2cd80%2C%200xc00646eca8)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fresolvers.go%3A3955%20%2B0x2ab%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.(*executionContext)._Version_status.func2(0x41b71a0%2C%200xc00ba9de60%2C%200x0%2C%200x0%2C%200xc00646ecf0%2C%200x46f37b)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fgenerated.go%3A37061%20%2B0x9c%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql%2Fexecutor.processExtensions.func3(0x41b71a0%2C%200xc00ba9de60%2C%200xc000f0e5e0%2C%200x4981c6%2C%200x62699afa%2C%200x4531192%2C%200x380fe3d33bcdba)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Fexecutor%2Fextensions.go%3A62%20%2B0x3a%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql%2Fexecutor.processExtensions.func6.1(0x41b71a0%2C%200xc00ba9de60%2C%200x5ad98e0%2C%200x7%2C%200x2ff1480%2C%200xc005120180)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Fexecutor%2Fextensions.go%3A91%20%2B0x42%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql%2Fhandler%2Fapollotracing.Tracer.InterceptField(0x41b71a0%2C%200xc00ba9de60%2C%200xc003009e90%2C%200x0%2C%200x0%2C%200x0%2C%200x0)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Fhandler%2Fapollotracing%2Ftracer.go%3A84%20%2B0x118%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql%2Fexecutor.processExtensions.func6(0x41b71a0%2C%200xc00ba9de60%2C%200xc000f0e5e0%2C%200x41b71a0%2C%200xc00ba9de60%2C%200x2%2C%200x2)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Fexecutor%2Fextensions.go%3A90%20%2B0xa3%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.(*executionContext)._Version_status(0xc00dd537c0%2C%200x41b71a0%2C%200xc004908900%2C%200xc002e34700%2C%200x0%2C%200x0%2C%200x0%2C%200xc00f22ba40%2C%200x0%2C%200x0)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fgenerated.go%3A37059%20%2B0x208%0Agithub.com%2Fevergreen-ci%2Fevergreen%2Fgraphql.(*executionContext)._Version.func1(0x0%2C%200x0)%0A%09github.com%2Fevergreen-ci%2Fevergreen%2Fgraphql%2Fgenerated.go%3A49239%20%2B0xc5%0Agithub.com%2F99designs%2Fgqlgen%2Fgraphql.(*FieldSet).Dispatch.func1(0xc006f4c460%2C%200xc0046f8480%2C%200x2%2C%200xc006f4c500)%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Ffieldset.go%3A42%20%2B0x2b%0Acreated%20by%20github.com%2F99designs%2Fgqlgen%2Fgraphql.(*FieldSet).Dispatch%0A%09github.com%2F99designs%2Fgqlgen%40v0.14.0%2Fgraphql%2Ffieldset.go%3A41%20%2B0x118%0A%22&display.page.search.mode=smart&dispatch.sample_ratio=1&workload_pool=standard_perf&sid=1651091576.407956

Reply:
i just checked and it looks like it could be empty if you never actually schedule a patch and in that case we would just want to ignore it so i think it makes sense to just ignore the cp if that's the case 

we can probably make a separate check and continue if cpVersion == nil